core/bindings.rs

```diff
@@ -453,17 +461,56 @@ fn opcall_async<'s>(
       state.pending_ops.push(fut);
       state.have_unpolled_ops = true;
     }
-    Op::AsyncUnref(fut) => {
-      state.op_state.borrow().tracker.track_unref(op_id);
-      state.pending_unref_ops.push(fut);
-      state.have_unpolled_ops = true;
-    }
     Op::NotFound => {
       throw_type_error(scope, format!("Unknown op id: {}", op_id));
     }
   }
 }
 
+fn ref_op<'s>(
+  scope: &mut v8::HandleScope<'s>,
+  args: v8::FunctionCallbackArguments,
+  _rv: v8::ReturnValue,
+) {
+  let state_rc = JsRuntime::state(scope);
+  let mut state = state_rc.borrow_mut();
+
+  let promise_id = match v8::Local::<v8::Integer>::try_from(args.get(0))
+    .map(|l| l.value() as PromiseId)
+    .map_err(Error::from)
+  {
+    Ok(promise_id) => promise_id,
+    Err(err) => {
+      throw_type_error(scope, format!("invalid promise id: {}", err));
+      return;
+    }
+  };
+
+  state.unrefed_ops.remove(&promise_id);
+}
+
+fn unref_op<'s>(
+  scope: &mut v8::HandleScope<'s>,
+  args: v8::FunctionCallbackArguments,
+  _rv: v8::ReturnValue,
+) {
+  let state_rc = JsRuntime::state(scope);
+  let mut state = state_rc.borrow_mut();
+
+  let promise_id = match v8::Local::<v8::Integer>::try_from(args.get(0))
+    .map(|l| l.value() as PromiseId)
+    .map_err(Error::from)
+  {
+    Ok(promise_id) => promise_id,
+    Err(err) => {
+      throw_type_error(scope, format!("invalid promise id: {}", err));
+      return;
+    }
+  };
+
+  state.unrefed_ops.insert(promise_id);
```

Comment:
Can we check that `promise_id` is in `state.pending_ops`?

Reply:
No, we'd have to iterate over all `pending_ops` killing the performance.

Reply:
Oh wait, this is `bindings.rs`, it's not a hot path. I guess I can add a check here.

Reply:
Actually not possible to do that, there's no way to inspect `promise_id` when iterating `pending_ops`

Reply:
This might be possible once @AaronO moves promise ring to Rust (#12314)

Reply:
In the meantime, what probably can be done is change the `unrefOp` JS function to check whether the promise ID exists and hasn't resolved, right?

It would be pretty bad if some careless code added an already-resolved promise ID to `unrefed_ops`, since all that matters is the length compared to `pending_ops`. But I suppose CI would catch that, though.

Reply:
> In the meantime, what probably can be done is change the unrefOp JS function to check whether the promise ID exists and hasn't resolved, right?

That would require to adding a wrapper around `Deno.core.unrefOp` in `core/01_core.js` just to verify the promise map.

I'm not sure if it's worth doing that at this moment; I think we should try this API in the wild and see if that's a problem.